python/src/tstools/unrFiles.py

```diff
@@ -0,0 +1,459 @@
+#!/usr/bin/env python3
+
+"""
+Module for downloading and handling UNR generated files.
+"""
+
+import requests as req
+import numpy as np
+
+from tstools import inputFileIO as ifio
+from tstools.util.convtime import convtime, yy_to_yyyy
+from tstools.util.nutils import msg_err
+
+########################################################################
+"""
+Constants
+"""
+
+ALL = 'all'
+BLANK_STR = '' 
+
+EQUIP_CHNG_CODE = 1
+EQ_CODE = 2
+
+JAN, FEB, MAR, APR, MAY = ['JAN','FEB','MAR','APR','MAY']
+JUN, JUL, AUG, SEP, OCT = ['JUN','JUL','AUG','SEP','OCT']
+NOV, DEC = ['NOV','DEC']
+
+########################################################################
+class UnrTxyz2:
```

Comment:
In general I don't think this is how I'd organize this class. If you're not careful in how you use it, you could end up with garbled results. For example, if I call `#fetch` twice in a row before I call `#write`, it appears that the contents of two separate files will be written to a single file, and they could have mismatched sites.

I like having the logic split into two sections, "how do I get data?" and "how do I write data?", but they shouldn't be able to mess each other up. This goes a bit to the comment I made about unintended side affects.

This would be a fun class to pair on at some point, maybe if we get some time while we're in San Fran.